mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -892,49 +291,54 @@ func resourceTemplateContractServiceGraphRead(d *schema.ResourceData, m interfac
 					if !contractCont.Exists("serviceGraphRelationship") {
 						return fmt.Errorf("No service graph found")
 					} else {
-
+						// Template Contract Service Graph configurations
 						graphRelation := contractCont.S("serviceGraphRelationship")
-
 						graphRef := models.StripQuotes(graphRelation.S("serviceGraphRef").String())
 						tokens := strings.Split(graphRef, "/")
-						d.Set("service_graph_name", tokens[len(tokens)-1])
-						if _, ok := d.GetOk("service_graph_schema_id"); ok {
-							d.Set("service_graph_schema_id", tokens[len(tokens)-5])
+
+						d.Set("service_graph_name", tokens[6])
+
+						if _, ok := d.GetOk("service_graph_schema_id"); !ok {
+							d.Set("service_graph_schema_id", tokens[2])
 						} else {
-							d.Set("service_graph_schema_id", "")
+							d.Set("service_graph_schema_id", d.Get("service_graph_schema_id"))
 						}
-						if _, ok := d.GetOk("service_graph_template_name"); ok {
-							d.Set("service_graph_template_name", tokens[len(tokens)-3])
+
+						if _, ok := d.GetOk("service_graph_template_name"); !ok {
+							d.Set("service_graph_template_name", tokens[4])
 						} else {
-							d.Set("service_graph_template_name", "")
+							d.Set("service_graph_template_name", d.Get("service_graph_template_name"))
 						}
 
+						// Template Contract Service Graph Node configurations
 						nodeCount, err := graphRelation.ArrayCount("serviceNodesRelationship")
 						if err != nil {
 							return err
 						}
+						nodeInterface := d.Get("node_relationship")
+
 						for k := 0; k < nodeCount; k++ {
 							relationMap := make(map[string]interface{})
 							node, err := graphRelation.ArrayElement(k, "serviceNodesRelationship")
 							if err != nil {
 								return err
 							}
-							nodeRef := models.StripQuotes(node.S("serviceNodeRef").String())
-							tokensNode := strings.Split(nodeRef, "/")
-							relationMap["node_name"] = tokensNode[len(tokensNode)-1]
-
-							nodeInterface := d.Get("node_relationship")
 
 							probdRef := models.StripQuotes(node.S("providerConnector", "bdRef").String())
 							probdRefTokens := strings.Split(probdRef, "/")
 							relationMap["provider_connector_bd_name"] = probdRefTokens[len(probdRefTokens)-1]
 
-							if checkNodeAttr(nodeInterface, "provider_connector_bd_schema_id", k) {
+							if importFlag {
+								relationMap["provider_connector_bd_schema_id"] = probdRefTokens[len(probdRefTokens)-5]
+							} else if checkNodeAttr(nodeInterface, "provider_connector_bd_schema_id", k) {
```

Comment:
shoudl we combine the conditional?
```suggestion
							if importFlag || checkNodeAttr(nodeInterface, "provider_connector_bd_schema_id", k) {
```